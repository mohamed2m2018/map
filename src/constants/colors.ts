const colors = {
  lightTheme: {
    primary: '#ffffff',
    secondary: '#000000',
    tabBarBackground: '#ffffff',
    grey: 'grey',
  },
  darkTheme: {
    primary: '#444444',
    secondary: '#ffffff',
    tabBarBackground: '#000000',
    grey: 'grey',
  },
};

export interface IColors {
  primary: string;
  secondary: string;
  tabBarBackground: string;
  grey: string;
}

export default colors;
