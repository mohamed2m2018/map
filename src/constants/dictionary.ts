const dictionary = {
  somethingWrong: 'Something wrong happened',
  choosePlace: 'Please choose a place first',
  noAccess: "Can't access location",
};

export default dictionary;
