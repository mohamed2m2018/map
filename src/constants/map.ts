export const DARK_MODE_STYLE = [
  {
    elementType: 'geometry',
    stylers: [
      {
        color: '#212121',
      },
    ],
  },
  {
    elementType: 'labels.icon',
    stylers: [
      {
        visibility: 'off',
      },
    ],
  },
  {
    elementType: 'labels.text.fill',
    stylers: [
      {
        color: '#757575',
      },
    ],
  },
  {
    elementType: 'labels.text.stroke',
    stylers: [
      {
        color: '#212121',
      },
    ],
  },
  {
    featureType: 'administrative',
    elementType: 'geometry',
    stylers: [
      {
        color: '#757575',
      },
    ],
  },
  {
    featureType: 'administrative.country',
    elementType: 'labels.text.fill',
    stylers: [
      {
        color: '#9e9e9e',
      },
    ],
  },
  {
    featureType: 'administrative.land_parcel',
    stylers: [
      {
        visibility: 'off',
      },
    ],
  },
  {
    featureType: 'administrative.locality',
    elementType: 'labels.text.fill',
    stylers: [
      {
        color: '#bdbdbd',
      },
    ],
  },
  {
    featureType: 'poi',
    elementType: 'labels.text.fill',
    stylers: [
      {
        color: '#757575',
      },
    ],
  },
  {
    featureType: 'poi.park',
    elementType: 'geometry',
    stylers: [
      {
        color: '#181818',
      },
    ],
  },
  {
    featureType: 'poi.park',
    elementType: 'labels.text.fill',
    stylers: [
      {
        color: '#616161',
      },
    ],
  },
  {
    featureType: 'poi.park',
    elementType: 'labels.text.stroke',
    stylers: [
      {
        color: '#1b1b1b',
      },
    ],
  },
  {
    featureType: 'road',
    elementType: 'geometry.fill',
    stylers: [
      {
        color: '#2c2c2c',
      },
    ],
  },
  {
    featureType: 'road',
    elementType: 'labels.text.fill',
    stylers: [
      {
        color: '#8a8a8a',
      },
    ],
  },
  {
    featureType: 'road.arterial',
    elementType: 'geometry',
    stylers: [
      {
        color: '#373737',
      },
    ],
  },
  {
    featureType: 'road.highway',
    elementType: 'geometry',
    stylers: [
      {
        color: '#3c3c3c',
      },
    ],
  },
  {
    featureType: 'road.highway.controlled_access',
    elementType: 'geometry',
    stylers: [
      {
        color: '#4e4e4e',
      },
    ],
  },
  {
    featureType: 'road.local',
    elementType: 'labels.text.fill',
    stylers: [
      {
        color: '#616161',
      },
    ],
  },
  {
    featureType: 'transit',
    elementType: 'labels.text.fill',
    stylers: [
      {
        color: '#757575',
      },
    ],
  },
  {
    featureType: 'water',
    elementType: 'geometry',
    stylers: [
      {
        color: '#000000',
      },
    ],
  },
  {
    featureType: 'water',
    elementType: 'labels.text.fill',
    stylers: [
      {
        color: '#3d3d3d',
      },
    ],
  },
];

export const PLACES_DATA = [
  {
    id: 1,
    name: 'Costa Coffee',
    description: 'The best coffee',
    imageOnMap: 'https://i.ibb.co/TKyfJYn/a.png',
    thumbnailImage:
      'https://logos-download.com/wp-content/uploads/2016/05/Costa_Coffee_logo_logotype-700x700.png',
    latitudeDifference: 0.001,
    longitudeDifference: -0.001,
  },
  {
    id: 2,
    name: 'Ice Cream Shop',
    description: 'Your favorite ice cream',
    imageOnMap: 'https://i.ibb.co/ZLTFw47/aa-1.png',
    thumbnailImage:
      'https://img.freepik.com/premium-vector/ice-cream-logo-design-illustration_9845-323.jpg',
    latitudeDifference: 0.002,
    longitudeDifference: 0.001,
  },
  {
    id: 3,
    name: 'Macdonalds',
    description: 'Best burger',
    imageOnMap: 'https://i.ibb.co/yBCgzwg/burger2.png',
    thumbnailImage:
      'https://mpng.subpng.com/20190227/exp/kisspng-logo-computer-icons-scalable-vector-graphics-clip-filemcdonaldampapossshadow-svg-wikipedia-5c775acd244151.2332977515513259011485.jpg',
    latitudeDifference: -0.002,
    longitudeDifference: 0.002,
  },
];
