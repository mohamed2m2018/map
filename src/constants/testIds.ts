export const TEST_IDS = {
  TEST_ID_SHOP_NAME: 'TEST_ID_SHOP_NAME',
  TEST_ID_SHOP_DESCRIPTION: 'TEST_ID_SHOP_DESCRIPTION',
  TEST_ID_IMAGE_SHOP: 'TEST_ID_IMAGE_SHOP',
  MAP_BUTTON_TOUCHABLE: 'MAP_BUTTON_TOUCHABLE',
};
