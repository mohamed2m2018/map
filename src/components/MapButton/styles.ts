import {IColors} from './../../constants/colors';
import {StyleSheet} from 'react-native';

const makeStyles = (colors: IColors) =>
  StyleSheet.create({
    touchableContainer: {
      width: 45,
      height: 45,
      backgroundColor: colors.primary,
      borderRadius: 22.5,
      justifyContent: 'center',
      alignItems: 'center',
    },
  });

export default makeStyles;
