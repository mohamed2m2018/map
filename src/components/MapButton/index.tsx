import React, {useContext} from 'react';
import {TouchableOpacity} from 'react-native';
import {TEST_IDS} from '../../constants/testIds';
import {ThemeContext} from '../../context/ThemeContext';
import AppIcon, {AppIconProps} from '../AppIcon';
import makeStyles from './styles';

interface Props extends AppIconProps {
  onPress: () => void;
}

const MapButton: React.FC<Props> = ({type, onPress, name}) => {
  const {colors} = useContext(ThemeContext);
  const styles = makeStyles(colors);
  return (
    <TouchableOpacity
      testID={TEST_IDS.MAP_BUTTON_TOUCHABLE}
      onPress={onPress}
      style={styles.touchableContainer}>
      <AppIcon type={type} name={name} size={25} color={colors.secondary} />
    </TouchableOpacity>
  );
};

export default MapButton;
