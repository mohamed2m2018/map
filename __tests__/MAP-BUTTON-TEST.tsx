import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer, {act} from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';
import {TEST_IDS} from '../src/constants/testIds';
import MapButton from '../src/components/MapButton';

let props = {
  name: 'swap',
  onPress: () => {},
};

describe('Map button', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('renders correctly', () => {
    const screenToBeSnapped = renderer
      .create(<MapButton {...props} />)
      .toJSON();
    expect(screenToBeSnapped).toMatchSnapshot();
  });

  it('fires onPress prop when clicked on', async () => {
    const MethodToBeSent = jest.fn(() => {});
    props.onPress = MethodToBeSent;
    const {getByTestId} = render(<MapButton {...props} />);
    const touchable = getByTestId(TEST_IDS.MAP_BUTTON_TOUCHABLE);
    await act(async () => {
      fireEvent.press(touchable);
    });
    expect(MethodToBeSent).toHaveBeenCalled();
  });
});
