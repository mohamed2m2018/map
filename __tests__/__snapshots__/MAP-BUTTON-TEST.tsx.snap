// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map button renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 22.5,
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "width": 45,
    }
  }
  testID="MAP_BUTTON_TOUCHABLE"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 25,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ?
  </Text>
</View>
`;
